consensus/istanbul/backend/handler.go

```diff
@@ -409,14 +407,24 @@ func (sb *Backend) readValidatorHandshakeMessage(peer consensus.Peer) (bool, err
 		return false, errors.New("Incorrect node in enodeCertificate")
 	}
 
-	block := sb.currentBlock()
-	valSet := sb.getValidators(block.Number().Uint64(), block.Hash())
-	if !valSet.ContainsByAddress(sb.ValidatorAddress()) {
-		logger.Trace("This validator is not in the validator set")
+	// Check if the peer is within the validator conn set.
+	validatorConnSet := sb.retrieveCachedValidatorConnSet()
```

Comment:
What's the need for the new function `retrieveCachedValidatorConnSet`?  This function is updating the cache if there is a cache miss.

Reply:
This is because of some feedback in the handshake PR from @nategraf: https://github.com/celo-org/celo-blockchain/pull/873#discussion_r386743099

The rationale is that it'd be better to use some potentially stale data during the handshake to avoid risking a potentially long-blocking function. I know we managed to reduce the execution time for elections recently, do you think this is a non-issue now that we are using the validator con set?